gpuallocator/besteffort_policy.go

```diff
@@ -46,6 +46,33 @@ func (p *bestEffortPolicy) Allocate(available []*Device, required []*Device, siz
 		return []*Device{}
 	}
 
+	if len(required) > len(available) {
+		return []*Device{}
+	}
+
+	// Optimize for the case when we required is actually the `size`.
+	if size == len(required) {
+		if gpuPartitionContainsSetWithAll([][]*Device{available}, required) {
+			return required
+		} else {
+			return []*Device{}
+		}
+	}
+
+	// Optimize for the case when size == 1.
+	// We'll pick the device with the minimum sum of scores with available devices.
```

Comment:
We can discuss here.

Reply:
For requests with 1 device is the choice not irrellevant from the perspective of the application requesting the device? It may be that assigning the device with the lowest connectivity in this case provides some advantage since other more connected devices are available for future requests that require multple devices.

Do you have benchmarks to show whether this change improves performance at all?

Reply:
> It may be that assigning the device with the lowest connectivity in this case provides some advantage since other more connected devices are available for future requests that require multple devices.

Yes, exactly the case, benefit to the future request for multi-devices.

I have no benchmark yet, but I can provide one if needed, generally, this is not quite aimed for the performance but for the **application friendly and rational utilization** as you described above.